pkg/skaffold/build/kaniko/run.go

```diff
@@ -49,29 +46,26 @@ func (b *Builder) run(ctx context.Context, out io.Writer, artifact *latest.Artif
 	}
 	defer s.Cleanup(ctx)
 
+	kanikoArtifact := artifact.KanikoArtifact
 	// Create pod spec
 	args := []string{
-		"--dockerfile", artifact.DockerArtifact.DockerfilePath,
+		"--dockerfile", kanikoArtifact.DockerfilePath,
 		"--context", context,
 		"--destination", tag,
 		"-v", logLevel().String()}
-	args = append(args, b.AdditionalFlags...)
-	args = append(args, docker.GetBuildArgs(artifact.DockerArtifact)...)
+
+	// TODO: remove since AdditionalFlags will be deprecated (priyawadhwa@)
+	args = append(args, kanikoArtifact.AdditionalFlags...)
```

Comment:
let's put out a deprecation warning here into the log if `AdditionalFlags` are not empty?